File: solutions/0448-Find-All-Numbers-Disappeared-in-an-Array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
const findDisappearedNumbers = function(nums) {
  const nums2 = [...nums];
  let values = [];
  
  for (let i = 0; i< nums.length; i++) {
    nums2[nums[i] - 1] = nums[nums[i] - 1] * -1;
  }
  
  for (let i = 0; i< nums2.length; i++) {
    if (nums2[i] > 0) values.push(i + 1);
  }
  
  return values;
};

File: solutions/0448-Find-All-Numbers-Disappeared-in-an-Array.cpp
class Solution {
public:
  vector<int> findDisappearedNumbers(vector<int>& nums) {
    vector<int> v;
    if (nums.empty()) return v;
    
    int size = nums.size();
    sort(nums.begin(), nums.end());
    auto u = unique(nums.begin(), nums.end());
    nums.resize(distance(nums.begin(), u));
  
    for (int i = 1; i <= size; i++) {
      if (!binary_search(nums.begin(), nums.end(), i)) v.push_back(i);
    }
    
    return v;
  }
};
